src/main/java/com/openlattice/datastore/search/controllers/SearchController.java

```diff
@@ -148,9 +140,15 @@ public DataSearchResult searchEntitySetData( @RequestBody SearchConstraints sear
 
         validateSearch( searchConstraints );
 
+        final UUID[] entitySetIds = searchConstraints.getEntitySetIds();
+
+        final LinkedHashSet<UUID> uniqueEntitySetIds = Sets.newLinkedHashSetWithExpectedSize( entitySetIds.length );
+
+        Collections.addAll(uniqueEntitySetIds, entitySetIds);
+
         // check read on entity sets
         final var authorizedEntitySetIds = entitySetManager
-                .filterToAuthorizedNormalEntitySets( Set.of( searchConstraints.getEntitySetIds() ), READ_PERMISSION );
+                .filterToAuthorizedNormalEntitySets( uniqueEntitySetIds, READ_PERMISSION );
```

Comment:
These changes will be needed to make this compile https://github.com/openlattice/datastore/pull/520